### ISSUE TITLE
Generate classes/writers/readers for custom glTF extensions

### ISSUE THREAD

**Issue by mlavik1** [CesiumGS/cesium-native#805]
Hi, I'm wondering what would be the best way to add support for custom glTF extensions.

I suppose I could fork your [glTF repository](https://github.com/CesiumGS/glTF) and add my schema files there, and then also fork cesium-native add reference them in glTF.json - but that complicates things a bit more than I had hoped. Are there any other ways, or would you consider adding support for a custom extension folder and config file, so that I could call:
```
node index.js --schema [SAME PARAMS AS USUAL]... --customExtensions myProject/my_extension --customExtConfig myglTF.json
```

Thanks for reading!

**Comment by kring** [comment]
Yeah I think the two forks are the only option right now. Otherwise, as you suggested, the code generator needs to be extended to support externally-defined extensions. That'll likely require a bit of refactoring.

**Comment by mlavik1** [comment]
@kring Yeah. For generating the extension class files and JSON reader code it would maybe not be that much work, but the code generation output for the JSON writers would have to be changed quite a bit.. 

But thanks for answering! I've forked both repos now, and that worked fine for me :)
**So should we close this issue, or is it worth keeping it open?**

---

And just in case other users who want to do the same read this issue someday in future, this is what I did:
- Forked the [glTF repository](https://github.com/CesiumGS/glTF/tree/3d-tiles-next) ("tree/3d-tiles-next" branch for 3d-tiles next) and added my schema.json files to new a folder inside "extensions/2.0/Vendor"
- Forked [cesium-native](https://github.com/CesiumGS/cesium-native) and replaced the Cesium's glTF-repo URL with our own fork in [package.json](https://github.com/CesiumGS/cesium-native/blob/main/tools/generate-classes/package.json), and also modified [glTF.json](https://github.com/CesiumGS/cesium-native/blob/main/tools/generate-classes/glTF.json) by adding our extension to "extensions" and added class name overrides under "classes".
- Ran the `gltf-generate` command as described [here](https://github.com/catenda/cesium-native?tab=readme-ov-file#regenerate-gltf-and-3d-tiles-classes)

**Comment by kring** [comment]
> So should we close this issue, or is it worth keeping it open?

Let's leave it open. This is a use-case we'd like to support at some point.
